tb_plugin/torch_tb_profiler/profiler/module_op.py

```diff
@@ -0,0 +1,220 @@
+from copy import copy
+
+from .node import ModuleNode, ProfilerStepNode
+from .trace import EventTypes
+
+
+class Module:
+    def __init__(self, name, module_id):
+        self.name = name
+        self.module_id = module_id
+        self.children = []
+
+    def __hash__(self):
+        if self.children:
+            return hash((self.name, self.module_id, tuple(self.children)))
+        else:
+            return hash((self.name, self.module_id))
+
+    def __eq__(self, o) -> bool:
+        if not isinstance(o, Module):
+            return False
+
+        return (self.name == o.name and
+                self.module_id == o.module_id and
+                self.children == o.children)
+
+class ModuleStats:
+    def __init__(self, name, module_id):
+        self.name = name
+        self.module_id = module_id
+        self.occurences = 0
+        self.operators = 0
+        self.host_duration = 0
+        self.device_duration = 0
+        self.self_host_duration = 0
+        self.self_device_duration = 0
+
+    @property
+    def avg_host_duration(self):
+        return self.host_duration / self.occurences
+
+    @property
+    def avg_device_duration(self):
+        return self.device_duration / self.occurences
+
+
+def aggegate_module_view(tid2tree, events):
+    roots = _build_module_hierarchy(events)
+    modules = _get_module_list(tid2tree)
+    if modules and roots:
+        return _process_module_statistics(modules, roots)
+    else:
+        return None
+
+def _build_module_hierarchy(events):
+    '''Get the module hierarchy from the chome trace events
+    '''
+    python_events = [e for e in events if e.type in (EventTypes.PYTHON_FUNCTION, EventTypes.MODULE)]
+    python_events.sort(key=lambda e: e.python_id)
+    id_to_event = {e.python_id: e for e in python_events}
+
+    # Extract Python function topology.
+    children = {}
+    for e in python_events:
+        e_id = e.python_id
+        children.setdefault(e_id, [])
+        e_parent_id = e.python_parent_id
+        children.setdefault(e_parent_id, [])
+        children[e_parent_id].append(e_id)
+    function_leaves = [k for k, v in children.items() if not v]
+
+    # Convert Python function topology to Module topology.
+    #   This is a simple O(n) tree walking algorithm where we start from the leaves
+    #   and walk up, discarding any nodes which are not Module nodes.
+    module_parent_map = {}
+    seen = set()
+    for i in function_leaves:
```

Comment:
nit: usually good to have some tests around testable algorithms like this. not a blocker

Reply:
will add testcases when it is ready to review